File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
using namespace std;

int solution(vector<vector<int>> sizes) {
    int mw = 0;
    int mh = 0;
    for (auto s : sizes) {
        mw = max(mw, max(s[0], s[1]));
        mh = max(mh, min(s[0], s[1]));
    }
    return mw * mh;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    # 가로, 세로 중 큰 값을 가로로 몰고 작은 값을 세로로 몰아서 최대값 구하기
    w_max, h_max = 0, 0
    for w, h in sizes:
        if w < h:
            w, h = h, w
        w_max = max(w_max, w)
        h_max = max(h_max, h)
        
    return w_max * h_max